### PR TITLE
Display word-descriptive log level while enabling structured log formatting in json

### PR DESCRIPTION
**Pull request by wheatdog**
To address #57355

As example stated [here](https://github.com/ClickHouse/ClickHouse/blob/v23.10.6.60-stable/programs/server/config.xml#L66). ClickHouse should display word-descriptive log level value instead of numeric format.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Display word-descriptive log level while enabling structured log formatting in json

### Documentation entry for user-facing changes

no need

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by wheatdog** [comment]
I've tested this patch locally. It would spit out logs like following
```
> docker run --platform linux/amd64 --rm myclickhouse 
                                
Processing configuration file '/etc/clickhouse-server/config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/99-my-settings.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
Logging information to /var/log/clickhouse-server/clickhouse-server.log
Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
{"date_time":"1705559931.130354","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"SentryWriter","message":"Sending crash reports is disabled","source_file":"src\/Daemon\/SentryWriter.cpp; void SentryWriter::initialize(Poco::Util::LayeredConfiguration &)","source_line":"131"}
{"date_time":"1705559931.318622","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Starting ClickHouse 24.1.1.1 (revision: 54482, git hash: 5775596c4f964523cfaf59a041d7807f98944f7f, build id: 1C2E6B09B3D7BF5D41CCB36B736825FA5C134F6B), PID 1","source_file":"","source_line":"0"}
{"date_time":"1705559931.320601","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"starting up","source_file":"","source_line":"0"}
{"date_time":"1705559931.321271","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"OS name: Linux, version: 6.5.13-orbstack-00121-ge428743e4e98, architecture: x86_64","source_file":"programs\/server\/Server.cpp; virtual void DB::Server::initialize(Poco::Util::Application &)","source_line":"425"}
{"date_time":"1705559931.493091","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Available RAM: 7.75 GiB; physical cores: 8; logical cores: 8.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"712"}
```

**Review by yariks5s — CHANGES_REQUESTED** [review]
@wheatdog, can you please add some tests with different levels?

**Comment by wheatdog** [comment]
@yariks5s, thanks for the review. Do you mean manually test it and check results with more logger levels?

Here are some more examples.

```
❯ docker run --platform linux/amd64 --rm myclickhouse                                 
Processing configuration file '/etc/clickhouse-server/config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/99-my-settings.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
Logging information to /var/log/clickhouse-server/clickhouse-server.log
Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
{"date_time":"1705559931.130354","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"SentryWriter","message":"Sending crash reports is disabled","source_file":"src\/Daemon\/SentryWriter.cpp; void SentryWriter::initialize(Poco::Util::LayeredConfiguration &)","source_line":"131"}
{"date_time":"1705559931.318622","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Starting ClickHouse 24.1.1.1 (revision: 54482, git hash: 5775596c4f964523cfaf59a041d7807f98944f7f, build id: 1C2E6B09B3D7BF5D41CCB36B736825FA5C134F6B), PID 1","source_file":"","source_line":"0"}
{"date_time":"1705559931.320601","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"starting up","source_file":"","source_line":"0"}
{"date_time":"1705559931.321271","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"OS name: Linux, version: 6.5.13-orbstack-00121-ge428743e4e98, architecture: x86_64","source_file":"programs\/server\/Server.cpp; virtual void DB::Server::initialize(Poco::Util::Application &)","source_line":"425"}
{"date_time":"1705559931.493091","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Available RAM: 7.75 GiB; physical cores: 8; logical cores: 8.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"712"}
{"date_time":"1705559931.503874","thread_name":"","thread_id":"1","level":"Warning","query_id":"","logger_name":"Context","message":"Delay accounting is not enabled, OSIOWaitMicroseconds will not be gathered. Check \/proc\/sys\/kernel\/task_delayacct","source_file":"","source_line":"0"}
{"date_time":"1705559931.563274","thread_name":"","thread_id":"1","level":"Warning","query_id":"","logger_name":"Application","message":"Integrity check of the executable skipped because the reference checksum could not be read.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"902"}
{"date_time":"1705559931.563733","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"It looks like the process has no CAP_IPC_LOCK capability, binary mlock will be disabled. It could happen due to incorrect ClickHouse package installation. You could resolve the problem manually with 'sudo setcap cap_ipc_lock=+ep \/usr\/bin\/clickhouse'. Note that it will not work on 'nosuid' mounted filesystems.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"979"}
{"date_time":"1705559931.598976","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Lowered uncompressed cache size to 3.87 GiB because the system has limited RAM","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"1189"}
{"date_time":"1705559931.601272","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Lowered mark cache size to 3.87 GiB because the system has limited RAM","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"1199"}
{"date_time":"1705559931.602110","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Lowered index mark cache size to 3.87 GiB because the system has limited RAM","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"1219"}
{"date_time":"1705559931.643508","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Setting max_server_memory_usage was set to 6.97 GiB (7.75 GiB available * 0.90 max_server_memory_usage_to_ram_ratio)","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()(ConfigurationPtr, bool) const","source_line":"1293"}
{"date_time":"1705559931.644882","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Setting merges_mutations_memory_usage_soft_limit was set to 3.87 GiB (7.75 GiB available * 0.50 merges_mutations_memory_usage_to_ram_ratio)","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()(ConfigurationPtr, bool) const","source_line":"1321"}
{"date_time":"1705559931.645166","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Merges and mutations memory limit is set to 3.87 GiB","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()(ConfigurationPtr, bool) const","source_line":"1334"}
{"date_time":"1705559931.688196","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"BackgroundSchedulePool\/BgBufSchPool","message":"Create BackgroundSchedulePool with 16 threads","source_file":"src\/Core\/BackgroundSchedulePool.cpp; DB::BackgroundSchedulePool::BackgroundSchedulePool(size_t, CurrentMetrics::Metric, CurrentMetrics::Metric, const char *)","source_line":"163"}
{"date_time":"1705559931.706573","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"BackgroundSchedulePool\/BgSchPool","message":"Create BackgroundSchedulePool with 512 threads","source_file":"src\/Core\/BackgroundSchedulePool.cpp; DB::BackgroundSchedulePool::BackgroundSchedulePool(size_t, CurrentMetrics::Metric, CurrentMetrics::Metric, const char *)","source_line":"163"}
{"date_time":"1705559931.887620","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"BackgroundSchedulePool\/BgMBSchPool","message":"Create BackgroundSchedulePool with 16 threads","source_file":"src\/Core\/BackgroundSchedulePool.cpp; DB::BackgroundSchedulePool::BackgroundSchedulePool(size_t, CurrentMetrics::Metric, CurrentMetrics::Metric, const char *)","source_line":"163"}
{"date_time":"1705559931.892537","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"BackgroundSchedulePool\/BgDistSchPool","message":"Create BackgroundSchedulePool with 16 threads","source_file":"src\/Core\/BackgroundSchedulePool.cpp; DB::BackgroundSchedulePool::BackgroundSchedulePool(size_t, CurrentMetrics::Metric, CurrentMetrics::Metric, const char *)","source_line":"163"}
{"date_time":"1705559931.904072","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"CertificateReloader","message":"One of paths is empty. Cannot apply new configuration for certificates. Fill all paths and try again.","source_file":"src\/Server\/CertificateReloader.cpp; void DB::CertificateReloader::tryLoad(const Poco::Util::AbstractConfiguration &)","source_line":"72"}
{"date_time":"1705559931.923964","thread_name":"","thread_id":"1","level":"Warning","query_id":"","logger_name":"Application","message":"Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, DNS error: EAI: Address family for hostname not supported (version 24.1.1.1). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0<\/listen_host> . Example for disabled IPv4: <listen_host>::<\/listen_host>","source_file":"programs\/server\/Server.cpp; void DB::Server::createServer(Poco::Util::AbstractConfiguration &, const std::string &, const char *, bool, bool, std::vector<ProtocolServerAdapter> &, CreateServerFunc &&) const","source_line":"354"}
{"date_time":"1705559931.934829","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Listening for replica communication (interserver): http:\/\/0.0.0.0:9009","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"1629"}
{"date_time":"1705559931.995050","thread_name":"","thread_id":"1","level":"Warning","query_id":"","logger_name":"Access(local_directory)","message":"File \/var\/lib\/clickhouse\/access\/users.list doesn't exist","source_file":"src\/Access\/DiskAccessStorage.cpp; bool DB::DiskAccessStorage::readLists()","source_line":"240"}
{"date_time":"1705559931.995162","thread_name":"","thread_id":"1","level":"Warning","query_id":"","logger_name":"Access(local_directory)","message":"Recovering lists in directory \/var\/lib\/clickhouse\/access\/","source_file":"src\/Access\/DiskAccessStorage.cpp; DB::DiskAccessStorage::DiskAccessStorage(const String &, const String &, AccessChangesNotifier &, bool, bool)","source_line":"189"}
{"date_time":"1705559932.015613","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Context","message":"Initialized background executor for merges and mutations with num_threads=16, num_tasks=32, scheduling_policy=round_robin","source_file":"src\/Interpreters\/Context.cpp; void DB::Context::initializeBackgroundExecutorsIfNeeded()","source_line":"4848"}
{"date_time":"1705559932.018217","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Context","message":"Initialized background executor for move operations with num_threads=8, num_tasks=8","source_file":"src\/Interpreters\/Context.cpp; void DB::Context::initializeBackgroundExecutorsIfNeeded()","source_line":"4858"}
{"date_time":"1705559932.024451","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Context","message":"Initialized background executor for fetches with num_threads=16, num_tasks=16","source_file":"src\/Interpreters\/Context.cpp; void DB::Context::initializeBackgroundExecutorsIfNeeded()","source_line":"4873"}
{"date_time":"1705559932.027013","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Context","message":"Initialized background executor for common operations (e.g. clearing old parts) with num_threads=8, num_tasks=8","source_file":"src\/Interpreters\/Context.cpp; void DB::Context::initializeBackgroundExecutorsIfNeeded()","source_line":"4883"}
{"date_time":"1705559932.048401","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"DNSCacheUpdater","message":"Update period 15 seconds","source_file":"src\/Interpreters\/DNSCacheUpdater.cpp; void DB::DNSCacheUpdater::start()","source_line":"48"}
{"date_time":"1705559932.049099","thread_name":"","thread_id":"1","level":"Information","query_id":"","logger_name":"Application","message":"Loading metadata from \/var\/lib\/clickhouse\/","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"1738"}
```


```
❯ docker exec -it epic_bhaskara /bin/bash
root@d0fea3c8fb70:/# /entrypoint.sh
ClickHouse Database directory appears to contain a database; Skipping initialization
Processing configuration file '/etc/clickhouse-server/config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/99-my-settings.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
Logging information to /var/log/clickhouse-server/clickhouse-server.log
Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
{"date_time":"1705581359.701918","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"SentryWriter","message":"Sending crash reports is disabled","source_file":"src\/Daemon\/SentryWriter.cpp; void SentryWriter::initialize(Poco::Util::LayeredConfiguration &)","source_line":"131"}
{"date_time":"1705581359.739911","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"Starting ClickHouse 24.1.1.1 (revision: 54482, git hash: 5775596c4f964523cfaf59a041d7807f98944f7f, build id: 1C2E6B09B3D7BF5D41CCB36B736825FA5C134F6B), PID 2185","source_file":"","source_line":"0"}
{"date_time":"1705581359.740603","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"starting up","source_file":"","source_line":"0"}
{"date_time":"1705581359.740900","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"OS name: Linux, version: 6.5.13-orbstack-00121-ge428743e4e98, architecture: x86_64","source_file":"programs\/server\/Server.cpp; virtual void DB::Server::initialize(Poco::Util::Application &)","source_line":"425"}
{"date_time":"1705581359.806430","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"Available RAM: 7.75 GiB; physical cores: 8; logical cores: 8.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"712"}
{"date_time":"1705581359.812241","thread_name":"","thread_id":"2185","level":"Warning","query_id":"","logger_name":"Context","message":"Delay accounting is not enabled, OSIOWaitMicroseconds will not be gathered. Check \/proc\/sys\/kernel\/task_delayacct","source_file":"","source_line":"0"}
{"date_time":"1705581359.831660","thread_name":"","thread_id":"2185","level":"Warning","query_id":"","logger_name":"Application","message":"Integrity check of the executable skipped because the reference checksum could not be read.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"902"}
{"date_time":"1705581359.831810","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"It looks like the process has no CAP_IPC_LOCK capability, binary mlock will be disabled. It could happen due to incorrect ClickHouse package installation. You could resolve the problem manually with 'sudo setcap cap_ipc_lock=+ep \/usr\/bin\/clickhouse'. Note that it will not work on 'nosuid' mounted filesystems.","source_file":"programs\/server\/Server.cpp; virtual int DB::Server::main(const std::vector<std::string> &)","source_line":"979"}
{"date_time":"1705581359.834262","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"StatusFile","message":"Status file \/var\/lib\/clickhouse\/status already exists - unclean restart. Contents:\nPID: 1\nStarted at: 2024-01-18 06:38:51\nRevision: 54482\n","source_file":"src\/Common\/StatusFile.cpp; DB::StatusFile::StatusFile(std::string, FillFunction)","source_line":"59"}
{"date_time":"1705581359.844384","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"Shutting down storages.","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()() const","source_line":"739"}
{"date_time":"1705581359.847850","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"Waiting for background threads","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()() const","source_line":"726"}
{"date_time":"1705581359.847898","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"Background threads finished in 0 ms","source_file":"programs\/server\/Server.cpp; auto DB::Server::main(const std::vector<std::string> &)::(anonymous class)::operator()() const","source_line":"728"}
{"date_time":"1705581359.875203","thread_name":"","thread_id":"2185","level":"Error","query_id":"","logger_name":"Application","message":"Code: 76. DB::Exception: Cannot lock file \/var\/lib\/clickhouse\/status. Another server instance in same directory is already running. (CANNOT_OPEN_FILE), Stack trace (when copying this message, always include the lines below):\n\n0. .\/contrib\/llvm-project\/libcxx\/include\/exception:134: Poco::Exception::Exception(String const&, int) @ 0x00000000134e0132 in \/usr\/bin\/clickhouse\n1. .\/build_docker\/.\/src\/Common\/Exception.cpp:96: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000aba13b7 in \/usr\/bin\/clickhouse\n2. ? @ 0x0000000006449ae3 in \/usr\/bin\/clickhouse\n3. .\/build_docker\/.\/src\/Common\/StatusFile.cpp:75: DB::StatusFile::StatusFile(String, std::function<void (DB::WriteBuffer&)>) @ 0x000000000acaf4ed in \/usr\/bin\/clickhouse\n4. .\/contrib\/llvm-project\/libcxx\/include\/__functional\/function.h:818: ? @ 0x000000000ac23f7f in \/usr\/bin\/clickhouse\n5. .\/build_docker\/.\/base\/poco\/Util\/src\/Application.cpp:0: Poco::Util::Application::run() @ 0x000000001340eba6 in \/usr\/bin\/clickhouse\n6. .\/build_docker\/.\/programs\/server\/Server.cpp:414: DB::Server::run() @ 0x000000000ac1d315 in \/usr\/bin\/clickhouse\n7. .\/build_docker\/.\/base\/poco\/Util\/src\/ServerApplication.cpp:132: Poco::Util::ServerApplication::run(int, char**) @ 0x0000000013422611 in \/usr\/bin\/clickhouse\n8. .\/build_docker\/.\/programs\/server\/Server.cpp:0: mainEntryClickHouseServer(int, char**) @ 0x000000000ac1aed6 in \/usr\/bin\/clickhouse\n9. .\/build_docker\/.\/programs\/main.cpp:0: main @ 0x000000000642c439 in \/usr\/bin\/clickhouse\n10. ? @ 0x00007fffffdd0083\n11. ? @ 0x000000000642b3ae in \/usr\/bin\/clickhouse\n (version 24.1.1.1)","source_file":"src\/Common\/Exception.cpp; void DB::tryLogCurrentExceptionImpl(Poco::Logger *, const std::string &)","source_line":"222"}
{"date_time":"1705581359.875405","thread_name":"","thread_id":"2185","level":"Information","query_id":"","logger_name":"Application","message":"shutting down","source_file":"","source_line":"0"}
{"date_time":"1705581359.875981","thread_name":"#1","thread_id":"2225","level":"Information","query_id":"","logger_name":"BaseDaemon","message":"Stop SignalListener thread","source_file":"src\/Daemon\/BaseDaemon.cpp; virtual void SignalListener::run()","source_line":"237"}
```

BTW, this mapping (int->string) show follow
https://github.com/ClickHouse/ClickHouse/blob/b79ead9c847b9cf5077145611a1c13b8c45daa5b/base/poco/Foundation/src/PatternFormatter.cpp#L224-L244

**Comment by yariks5s** [comment]
I mean automated testing (one similar example is located [here](https://github.com/yariks5s/ClickHouse/commit/03194eaeb92c6225112db68b26e24b2f03a8116d#diff-737eb62e000f437502a9dddf4ba2cc5388e013ff0186d3f6a9bb2b74a1681053)), So you can create a new test or enhance existing ones to check if `level` setting has correct data

**Comment by appier-timliou** [comment]
Got it. Let me enhance that part.

**Comment by wheatdog** [comment]
Got it. Let me enhance that part.

**Comment by wheatdog** [comment]
@yariks5s I've enhance the related integration tests. 

I tried to follow [the guide](https://github.com/ClickHouse/ClickHouse/tree/master/tests/integration#running-with-runner-script) to run the corresponding integration test locally but I couldn't find `clickhouse-odbc-bridge`, which `runner` script required. It's not in my local build.

Could you guide me on how to properly do integration locally? Or could you review my test case and test it on CI. Thanks.

File: tests/integration/test_structured_logging_json/test.py [hunk 1]
```diff
@@ -80,7 +104,7 @@ def valiade_everything(config, node, config_type):
     logs = node.grep_in_log("").split("\n")
     return validate_logs(logs) and validate_log_config_relation(
         config, logs, config_type
-    )
+    ) and validate_log_level(config, logs)
```
Comment (yariks5s) [inline tests/integration/test_structured_logging_json/test.py:107]:
```suggestion
        and validate_log_level(config, logs)
    )

```

File: tests/integration/test_structured_logging_json/test.py [hunk 2]
```diff
@@ -80,7 +104,7 @@ def valiade_everything(config, node, config_type):
     logs = node.grep_in_log("").split("\n")
     return validate_logs(logs) and validate_log_config_relation(
         config, logs, config_type
```
Comment (yariks5s) [inline tests/integration/test_structured_logging_json/test.py:106]:
```suggestion
    return (
        validate_logs(logs)
        and validate_log_config_relation(config, logs, config_type)
```

File: tests/integration/test_structured_logging_json/test.py [hunk 3]
```diff
@@ -32,6 +32,30 @@ def is_json(log_json):
...
+        "Critical", 
+        "Error", 
+        "Warning", 
+        "Notice", 
+        "Information", 
+        "Debug", 
+        "Trace", 
+        "Test",
+    }
+
```
Comment (yariks5s) [inline tests/integration/test_structured_logging_json/test.py:50]:
```suggestion
    valid_level_values = {
        "Fatal",
        "Critical",
        "Error",
        "Warning",
        "Notice",
        "Information",
        "Debug",
        "Trace",
        "Test",
    }
```

**Comment by wheatdog** [comment]
Thanks @yariks5s, I've run black locally to reformat the code.

**Comment by wheatdog** [comment]
Hi @yariks5s, I noticed a several errors during CI. I'm not sure whether all of them are relevant to this PR. Could you guide me on how to proceed? Thanks.

The updated test case [test_structured_logging_json/test.py::test_structured_logging_json_format](https://s3.amazonaws.com/clickhouse-test-reports/58936/483bbaf14d16bb4b3871781c1bbc95eee7311a8f/integration_tests__release__[3_4]/integration_run_parallel5_0.log) is passed.

**Review by yariks5s — APPROVED** [review]

**Comment by yariks5s** [comment]
Can you sync your branch with master? I don't have rights to do this by myself.

**Comment by wheatdog** [comment]
Sure. I've merged the latest master branch into my feature branch.

**Pull request merged**